exchanges/graphcache/src/operations/query.ts

```diff
@@ -410,7 +410,7 @@ const resolveResolverResult = (
         joinKeys(key, `${i}`),
         select,
         // Get the inner previous data from prevData
-        prevData !== undefined ? prevData[i] : undefined,
+        prevData ? prevData[i] : undefined,
```

Comment:
I think something along the lines of this is safer to prevent false negatives this covers both null and undefined

```suggestion
        prevData != null ? prevData[i] : undefined,
```